src/components/vm/consoles/vnc.tsx

```diff
@@ -46,53 +49,60 @@ import { logDebug } from '../../../helpers.js';
 import { LaunchViewerButton, connection_address, ConsoleState } from './common';
 import { domainSendKey, domainAttachVnc, domainChangeVncSettings, domainGet } from '../../../libvirtApi/domain.js';
 
-import { VncConsole } from './VncConsole';
+import { VncConsole, VncCredentials } from './VncConsole';
+
+import store from '../../../store.js';
 
 const _ = cockpit.gettext;
-// https://github.com/torvalds/linux/blob/master/include/uapi/linux/input-event-codes.h
-const Enum = {
-    KEY_BACKSPACE: 14,
-    KEY_LEFTCTRL: 29,
-    KEY_LEFTALT: 56,
-    KEY_F1: 59,
-    KEY_F2: 60,
-    KEY_F3: 61,
-    KEY_F4: 62,
-    KEY_F5: 63,
-    KEY_F6: 64,
-    KEY_F7: 65,
-    KEY_F8: 66,
-    KEY_F9: 67,
-    KEY_F10: 68,
-    KEY_F11: 87,
-    KEY_F12: 88,
-    KEY_DELETE: 111,
-};
+
+export type VncSizeMode = "none" | "local" | "remote";
 
 export class VncState extends ConsoleState {
-    constructor() {
-        super();
-        this.sizeMode = "none";
-    }
+    sizeMode: VncSizeMode = "none";
+    clean_disconnect: boolean = true;
+    connection_failure: string | null = null;
 
-    setSizeMode(val) {
+    setSizeMode(val: VncSizeMode) {
         this.sizeMode = val;
         this.update();
     }
+
+    setConnected(val: boolean) {
+        this.connected = val;
+        this.connection_failure = null;
+        this.clean_disconnect = true;
+        this.update();
+    }
+
+    setDisconnected(clean: boolean) {
+        this.connected = false;
+        this.clean_disconnect = clean;
+        this.update();
+    }
+
+    setConnectionFailure(val: string) {
+        this.connection_failure = val;
+    }
 }
 
-const VncEditModal = ({ vm, inactive_vnc }) => {
-    const config_port = (inactive_vnc.port == -1) ? "" : (inactive_vnc.port || "");
+const VncEditModal = ({
+    vm,
+    inactive_vnc
+} : {
+    vm: VM,
+    inactive_vnc: VMGraphics,
+}) => {
+    const config_port = (Number(inactive_vnc.port) == -1) ? "" : (inactive_vnc.port || "");
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-2267-42a6eadf-20250808-101729-fedora-42-devel/Coverage/lcov/github-pr.diff.gcov.html)